### PR TITLE
[@types/applepayjs] Add applePayCapabilities method type

### PR DESCRIPTION
**Pull request by kalkv**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test /types/applepayjs`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/4440085-applepaycapabilities>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by martincostello** [comment]
[Apple Pay on the Web documentation](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/4440085-applepaycapabilities) is a better link than to a 20 minute video.

File: types/applepayjs/index.d.ts [hunk 1]
```diff
@@ -965,6 +973,37 @@ declare namespace ApplePayJS {
...
+    }
+
+    /**
+     * Information about whether the device supports Apple Pay and the possible payment credentials the person has provisioned in Wallet.
+     */
+    type PaymentCredentialStatus =
+        /**
+         * Confirms that the device supports Apple Pay and there’s at least one active payment credential in Wallet that qualifies for payments on the web. Show an Apple Pay button and offer Apple Pay as the primary, but not necessarily sole, payment option.
+         */
+        | "paymentCredentialsAvailable"
```
Comment (martincostello) [inline types/applepayjs/index.d.ts:993]:
```suggestion
        | "paymentCredentialStatusAvailable"
```

Comment (kalkson) [reply types/applepayjs/index.d.ts:993]: Are you sure about that? Documantation says clearly that this is `paymentCredentialsAvailable` status. Am I missing something?
image

Comment (martincostello) [reply types/applepayjs/index.d.ts:993]: Fun....

So, if you click on `[PaymentCredentialStatus](https://developer.apple.com/documentation/apple_pay_on_the_web/paymentcredentialstatus)` directly from [`applePayCapabilities`](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/4440085-applepaycapabilities), it takes you to [this document](https://developer.apple.com/documentation/apple_pay_on_the_web/paymentcredentialstatus) which states `paymentCredentialsAvailable`.

However, if you click on `PaymentCredentialStatusResponse` which takes you to [`PaymentCredentialStatusResponse`](https://developer.apple.com/documentation/apple_pay_on_the_web/paymentcredentialstatusresponse), then click on `paymentCredentialStatus`, it takes you to [`paymentCredentialStatus`](https://developer.apple.com/documentation/apple_pay_on_the_web/paymentcredentialstatusresponse/4440088-paymentcredentialstatus) which states `paymentCredentialStatusAvailable`.

{F49AB32F-53A6-4D55-ABAA-722037A78DB3}

Comment (kalkson) [reply types/applepayjs/index.d.ts:993]: Well, there is some inconsistency in the documentation then. I just checked actual result and it seems that correct value is `paymentCredentialsAvailable`.
image
I just submitted bug report to Apple about that.

**Review by martincostello — APPROVED** [review]

**Comment by kalkson** [comment]
Ready to merge

**Pull request merged**